test/type_test.ts

```diff
@@ -6,6 +6,11 @@ describe('types', () => {
      () => { expectTranslate('var x: foo.Bar;').to.equal(' foo . Bar x ;'); });
   it('drops type literals',
      () => { expectTranslate('var x: {x: string, y: number};').to.equal(' dynamic x ;'); });
+  it('translates string index signatures to dartisms', () => {
+    expectTranslate('var x: {[k: string]: any[]};')
```

Comment:
@alexeagle Would this PR allow `'var x: {size: number, [k: string]: any[]};'` ?

We should have test for that.